test/protocol/data/redis/check_redis.c

```diff
@@ -0,0 +1,540 @@
+#include <protocol/data/redis_include.h>
+
+#include <buffer/cc_buf.h>
+#include <cc_array.h>
+#include <cc_bstring.h>
+#include <cc_define.h>
+
+#include <check.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+/* define for each suite, local scope due to macro visibility rule */
+#define SUITE_NAME "redis"
+#define DEBUG_LOG  SUITE_NAME ".log"
+
+struct request *req;
+struct response *rsp;
+struct buf *buf;
+
+/*
+ * utilities
+ */
+static void
+test_setup(void)
+{
+    req = request_create();
+    rsp = response_create();
+    buf = buf_create();
+}
+
+static void
+test_reset(void)
+{
+    request_reset(req);
+    response_reset(rsp);
+    buf_reset(buf);
+}
+
+static void
+test_teardown(void)
+{
+    buf_destroy(&buf);
+    response_destroy(&rsp);
+    request_destroy(&req);
+}
+
+/**************
+ * test cases *
+ **************/
+
+/*
+ * token
+ */
+START_TEST(test_simple_string)
+{
+#define SERIALIZED "+foobar\r\n"
+#define STR "foobar"
+
+    struct element el;
+    int ret;
+    int len = sizeof(SERIALIZED) - 1;
+    char *pos;
+
+    test_reset();
+
+    /* compose */
+    el.type = ELEM_STR;
+    el.bstr = str2bstr(STR);
+    ret = compose_element(&buf, &el);
+    ck_assert_msg(ret == len, "bytes expected: %d, returned: %d", len, ret);
+    ck_assert_int_eq(cc_bcmp(buf->rpos, SERIALIZED, ret), 0);
+
+    /* parse */
+    pos = buf->rpos + 1;
```

Comment:
reusing `el` from the compose section makes this test less reliable as the production code may be not setting a property that is set by the previous portion of the test; break it up into two tests? use a second `el`-like variable instead? destroy its value in between?